Makefile

```diff
@@ -66,12 +76,41 @@ docker-clean:
 
 ### Tyk Go Plugin ########################################################################
 
+go/src/go.mod:
+	cd ./go/src ; \
+	go mod init tyk-plugin ; \
+	go get -d github.com/TykTechnologies/tyk@`git ls-remote https://github.com/TykTechnologies/tyk.git refs/tags/${TYK_VERSION} | awk '{print $$1;}'` ; \
+	go mod tidy ; \
+	go mod vendor
+
 # Builds Go plugin and moves it into local Tyk instance
 .PHONY: go-build
-go-build:
+go-build: go/src/go.mod
 	/bin/sh -c "cd ./go/src && go mod tidy && go mod vendor"
-	docker-compose run --rm tyk-plugin-compiler CustomGoPlugin.so
-	mv -f ./go/src/CustomGoPlugin*.so ./tyk/middleware/
+	docker-compose run --rm tyk-plugin-compiler FrontdoorCachePlugin.so _$$(date +%s)
+	mv -f ./go/src/FrontdoorCachePlugin*.so ./tyk/middleware/
+
+# Runs Go Linter
+lint:
+	/bin/sh -c "docker run --rm -v ${PWD}/go/src:/app -v ~/.cache/golangci-lint/v1.53.2:/root/.cache -w /app golangci/golangci-lint:v1.53.2 golangci-lint run"
+
+# Runs Go unit tests
+test:
+	/bin/sh -c "cd ./go/src && go test"
+
+# Run Go test coverage
+coverage:
+	mkdir -p /tmp/test-results ; \
+	cd ./go/src ; \
+	go test ./... -coverprofile coverage.out -covermode count ; \
+	grep -v tyk-plugin/tyk_util.go coverage.out > coverage.out.tmp ; \
+	mv coverage.out.tmp coverage.out ; \
+	go tool cover -func coverage.out ; \
+	go tool cover -html=coverage.out -o coverage.html ; \
+	mv coverage.out coverage.html /tmp/test-results ; \
+	totalCoverage=`go tool cover -func=/tmp/test-results/coverage.out | grep total | grep -Eo '[0-9]+\.[0-9]+'` ; \
+	echo "Total Coverage: $$totalCoverage %" ; \
+	rm -rf /tmp/test-results
```

Comment:
Added some convenience Makefile commands to run linter, test, and coverage commands.